internal/provider/function_direxists.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/hashicorp/terraform-plugin-framework/function"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var _ function.Function = &DirectoryExistsFunction{}
+
+type DirectoryExistsFunction struct{}
+
+func NewDirectoryExistsFunction() function.Function {
+	return &DirectoryExistsFunction{}
+}
+
+func (f *DirectoryExistsFunction) Metadata(ctx context.Context, req function.MetadataRequest, resp *function.MetadataResponse) {
+	resp.Name = "direxists"
+}
+
+func (f *DirectoryExistsFunction) Definition(ctx context.Context, req function.DefinitionRequest, resp *function.DefinitionResponse) {
+	resp.Definition = function.Definition{
+		Summary: "Determines whether a directory exists at a given path.",
+		Description: "Given a path string, will return true if the directory exists. " +
+			"This function works only with directories. If used with a regular file, FIFO, or other " +
```

Comment:
Is it worth mentioning/clarifying the behaviour with symlink'd dirs, I assume they're considered the same as regular dirs? What would constitute a FIFO in this context?

Reply:
I originally copied this over from `fileexists`, but I think it makes it more confusing because symlink could be considered a special mode lol.

What do you think of 275e9211d28e48859f76278b15801c59b9d5498e ?